assets/Mods/BPML_GenericFunctions/Scripts/main.lua

```diff
@@ -65,3 +65,148 @@ RegisterCustomEvent("ConstructPersistentObject", function(ParamContext, ParamCla
+    -- Type checking similar to PrintToModLoader
+    if HookName:get():type() ~= "FString" then error(string.format("RegisterHookFromBP Param #1 must be FString but was %s", HookName:get():type())) end
+    if FunctionToCall:get():type() ~= "FString" then error(string.format("RegisterHookFromBP Param #2 must be FString but was %s", FunctionToCall:get():type())) end
+
+    local HookName_AsString = HookName:get():ToString()
+
+    -- Safeguard for checking if the function hook target is actually valid
+    local OriginalFunction = StaticFindObject(HookName_AsString)
+    if OriginalFunction == nil or not OriginalFunction:IsValid() or OriginalFunction:type() ~= "UFunction" then
+        error(string.format("Tried to hook invalid function '%s'", HookName_AsString))
+    end
+
+    local FunctionName = FunctionToCall:get():ToString()
+    local CallbackFunction = Context:get()[FunctionName]
+    if CallbackFunction == nil or not CallbackFunction:IsValid() or CallbackFunction:type() ~= "UFunction" then
+        error(string.format("Function '%s' doesn't exist in '%s'", FunctionName, Context:get():GetFullName()))
+    end
+
+    if RegisteredBPHooks[HookName_AsString] == nil then
+        RegisteredBPHooks[HookName_AsString] = {}
+    end
+
+    local ExpectedContextClass = OriginalFunction:GetOuter()
+
+    if not ExpectedContextClass:IsValid() or not ExpectedContextClass:IsClass() then
+        error("Context Class was invalid or not a Class")
+    end
+
+    local OriginalTypes = {}
+    local CallbackTypes = {}
+
+    OriginalFunction:ForEachProperty(function(Property)
+        if Property:HasAnyPropertyFlags(
+            EPropertyFlags.CPF_ConstParm | 
+            EPropertyFlags.CPF_Parm | 
+            EPropertyFlags.CPF_OutParm | 
+            EPropertyFlags.CPF_ReturnParm
+        ) then
+            table.insert(OriginalTypes, Property)
+        end
+    end)
+
+    CallbackFunction:ForEachProperty(function(Property)
+        if Property:HasAnyPropertyFlags(
+            EPropertyFlags.CPF_ConstParm | 
+            EPropertyFlags.CPF_Parm | 
+            EPropertyFlags.CPF_OutParm | 
+            EPropertyFlags.CPF_ReturnParm
+        ) then
+            table.insert(CallbackTypes, Property)
+        end
+    end)
+
+    if #OriginalTypes+1 ~= #CallbackTypes then
+        error("Param count did not match!")
+    end
+
+    if CallbackTypes[1]:GetPropertyClass():GetFullName() ~= ExpectedContextClass:GetFullName() then
+        error(string.format("Context Class did not match the expected Class '%s'", ExpectedContextClass:GetFName():ToString()))
+    end
+
+    for i = 1, #OriginalTypes, 1 do
+        -- Using i+1 here to skip Context from our Callback function, maybe there's a cleaner/better way?
+        if ((CallbackTypes[i+1]:IsA(OriginalTypes[i]:GetClass())) or (type(OriginalTypes[i].IsFloatingPoint) == "function" and CallbackTypes[i+1]:IsFloatingPoint() and OriginalTypes[i]:IsFloatingPoint())) then
+            
```

Comment:
Is this empty branch intentional ?

Reply:
No, I'll fix it.

Reply:
I think it's intentional ? And should be inverted, which I think is this:
```lua
if (not CallbackTypes[i+1]:IsA(OriginalTypes[i]:GetClass())) and (not OriginalTypes[i].IsFloatingPoint or not CallbackTypes[i+1]:IsFloatingPoint() or not OriginalTypes[i]:IsFloatingPoint())) then
```

Reply:
Done.